compiler/rustc_log/src/lib.rs

```diff
@@ -45,20 +45,38 @@
 use std::env::{self, VarError};
 use std::fmt::{self, Display};
 use std::io::{self, IsTerminal};
+use std::str::FromStr;
 use tracing_core::{Event, Subscriber};
 use tracing_subscriber::filter::{Directive, EnvFilter, LevelFilter};
 use tracing_subscriber::fmt::{
     format::{self, FormatEvent, FormatFields},
     FmtContext,
 };
 use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::Layer;
 
-pub fn init_env_logger(env: &str) -> Result<(), Error> {
-    let filter = match env::var(env) {
+/// In contrast to `init_rustc_env_logger` this allows you to choose an env var
+/// other than `RUSTC_LOG`.
+pub fn init_env_logger(env: &str, ice_env: &str) -> Result<(), Error> {
+    let log_filter = match env::var(env) {
         Ok(env) => EnvFilter::new(env),
         _ => EnvFilter::default().add_directive(Directive::from(LevelFilter::WARN)),
     };
 
+    #[cfg(bootstrap)]
+    let error_filter = match env::var(ice_env) {
+        Ok(env) => tracing::Level::from_str(&env).unwrap(),
+        _ => tracing::Level::WARN,
+    };
+    #[cfg(not(bootstrap))]
```

Comment:
I tried doing some shortcuts to see if I could make it work the way I want prior to figuring our the right solution. This attempts to work around the issue with compiling the stage1 compiler with `RUSTC_ICE_LOG=TRACE` by only setting that default for the resulting stage1 binary, which would then compile the tests with that as the default level and produce nice SpanTraces in the tests below.

It didn't work. I'm investigating now.